central/sensor/telemetry/controller_impl.go

```diff
@@ -99,14 +100,21 @@ func (c *controller) streamingRequest(ctx context.Context, dataType central.Pull
 		go c.sendCancellation(requestID)
 	}()
 
+	// In case there's no progress regarding sensor sending telemetry response data,
+	// we should stop waiting.
+	progressTicker := time.NewTicker(progressTimeout)
+	defer progressTicker.Stop()
 	for {
 		var resp *central.TelemetryResponsePayload
 		select {
 		case <-ctx.Done():
 			return errors.Wrap(ctx.Err(), "context error")
 		case <-c.stopSig.Done():
 			return errors.Wrap(c.stopSig.Err(), "lost connection to sensor")
+		case <-progressTicker.C:
+			return errors.New("no data was received from sensor within expected period")
```

Comment:
It will be grate to also wrap this error message in the caller to augment it with central id. So it will be obvious what clusters are problematic.

Reply:
`no data` suggests there were no data which could not be true how about 
```suggestion
			return errors.Newf("sensor didn't sent any data in last %s", progressTimeout)
```